datafusion/src/physical_plan/expressions/binary.rs

```diff
@@ -334,6 +334,141 @@ fn modulus_decimal(left: &DecimalArray, right: &DecimalArray) -> Result<DecimalA
     Ok(decimal_builder.finish())
 }
 
+fn bitwise_and(left: ArrayRef, right: ArrayRef) -> Result<ArrayRef> {
```

Comment:
yeah, these kernels would really be nicer if they were in arrow (where we can probably leverage all the bitwise hackery). It is reasonable to start in datafusion however, and then move them to arrow

Reply:
> yeah, these kernels would really be nicer if they were in arrow (where we can probably leverage all the bitwise hackery). It is reasonable to start in datafusion however, and then move them to arrow

agree with you.
We can implement them in datafusion now, and migrate them to arrow kernel later.
But the intention of this pr is as an example for adding bitwise operation, I wish more contributors can be involved in datafusion and make the contribution from a `new contributor friendly issue`.